app/assets/stylesheets/components/modals.scss

```diff
@@ -1,32 +1,80 @@
 @import '../config/import';
 
-%modal-style {
-  display: inline-block;
-  width: 100%;
-}
-
 .crayons-modal {
-  @extend %modal-style;
-  @include generate-box(
-    $level: 3,
-    $bg: var(--card-bg),
-    $border: var(--card-color),
-    $color: var(--card-color)
-  );
-  padding: $su-7;
-  max-width: 640px;
+  position: fixed;
+  left: 0;
+  right: 0;
+  top: 0;
+  bottom: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  z-index: 500;
+  pointer-events: none;
 
-  &--s {
-    max-width: 480px;
-    padding: $su-6;
+  &__overlay {
+    background: var(--base-100);
+    opacity: 0.5;
+    position: absolute;
+    left: 0;
+    right: 0;
+    top: 0;
+    bottom: 0;
+    z-index: 600;
+    pointer-events: auto;
   }
 
-  &--l {
+  &__box {
+    @include generate-box(
+      $level: 3,
+      $bg: var(--card-bg),
+      $border: var(--card-color),
+      $color: var(--card-color)
+    );
+    display: flex;
+    flex-direction: column;
     max-width: 800px;
-    padding: $su-8;
+    width: 100%;
+    background: var(--card-bg); // TODO: ?
+    z-index: 700;
```

Comment:
Just making sure the z-indices merit being 500 up top and 700 here since we now have our utility classes for `z-index` which go up to `z-50`. If so, in another PR, we should audit the app for z-indices and see if they really need to be as high as they are set to.

Reply:
Yea z-indexes overall are magic numbers... Since it's a component I didn't really want to use utility-classes... But I 100% agree, we have to audit all our z-indexes at some point.